### PR TITLE
Anonymous auth 

### PR DESCRIPTION
**Pull request by yashwin1**
Added anonymous authentication

### CODE REVIEW

**Comment by yashwin1** [comment]
User id (uid) has been added to the api call

File: app/src/main/java/inspire2connect/inspire2connect/satyaChat/ChatActivity.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ public void onErrorResponse(VolleyError error) {
 
     private static String createURL(Context ctx, String text) {
         text = text.trim();
-        return ctx.getString(R.string.satya_chatbot_url) + "?query=" + text + "&lang=" + getCurLang();
+        return ctx.getString(R.string.satya_chatbot_url) + "?query=" + text + "&lang=" + getCurLang() + "&uid=" + firebaseUser.getUid();
```
Comment (chirag-jn) [inline app/src/main/java/inspire2connect/inspire2connect/satyaChat/ChatActivity.java:110]: typecast uid to string before storing as a failsafe.

File: app/src/main/java/inspire2connect/inspire2connect/utils/BaseActivity.java [hunk 2]
```diff
@@ -26,6 +28,10 @@
     public static final DatabaseReference whoReference = FirebaseDatabase.getInstance("https://washkaro-who.firebaseio.com").getReference();
     public static final DatabaseReference statsReference = FirebaseDatabase.getInstance("https://washkaro-stats.firebaseio.com").getReference();
 
+    // Firebase Authenticaiton
+    public static final FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
+    public static FirebaseUser firebaseUser = firebaseAuth.getCurrentUser();
```
Comment (chirag-jn) [inline app/src/main/java/inspire2connect/inspire2connect/utils/BaseActivity.java:33]: dont initialize here

File: app/src/main/java/inspire2connect/inspire2connect/home/WelcomeActivity.java [hunk 3]
```diff
@@ -83,6 +89,19 @@ protected void onCreate(Bundle savedInstanceState) {
...
         // Checking for first time launch - before calling setContentView()
         getSupportActionBar().hide();
+
+        // Firebase Anonymous Auth
+        if(firebaseUser == null){
+            firebaseAuth.signInAnonymously().addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                @Override
+                public void onComplete(@NonNull Task<AuthResult> task) {
+                    if (task.isSuccessful()){
+                        firebaseUser = firebaseAuth.getCurrentUser();
```
Comment (chirag-jn) [inline app/src/main/java/inspire2connect/inspire2connect/home/WelcomeActivity.java:99]: just perform the initialization steps here

**Pull request merged**